index.html

```diff
@@ -4292,6 +4306,144 @@ <h3>Proof Formats</h3>
       </section>
     </section>
 
+    <section class="normative">
+      <h2>Algorithms</h2>
+
+      <p>
+This section contains algorithms that can be used by implementations to
+perform common operations.
+      </p>
+
+      <section class="normative">
+        <h3>Verification</h3>
+
+        <p>
+This section contains an algorithm that <a>conforming verifier
+implementations</a> can run when verifying a <a>verifiable credential</a> or a
+<a>verifiable presentation</a>. This algorithm takes a byte sequence (bytes
+<var>inputBytes</var>) and a media type (string <var>inputMediaType</var>) as
+inputs, and produces an object (<var>verified</var>) that contains the
+following:
+        </p>
+
+        <ul>
+          <li>
+a verification status (boolean <var>verified</var>.<var>status</var>)
+          </li>
+          <li>
+a verified document (object <var>verified</var>.<var>document</var>)
+          </li>
+          <li>
+a media type (string <var>verified</var>.<var>mediaType</var>)
+          </li>
+          <li>
+zero or more optional warnings (array of <a>anomalyDescriptions</a> <var>verified</var>.<var>warnings</var>)
+          </li>
+          <li>
+zero or more optional errors (array of <a>anomalyDescriptions</a> <var>verified</var>.<var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>verified</var> to an empty object.
+          </li>
+          <li>
+Verify that the securing mechanism in use on <var>inputBytes</var> has not been
+tampered with, using <var>inputMediaType</var> as additional input to the
+algorithm. See Section <a href="#securing-mechanisms"></a>
+for details.
+          </li>
+          <li>
+If the verification of the securing mechanism is successful, set the following:
+            <ul>
+              <li>
+<var>verified</var>.<var>status</var> to `true`
+              </li>
+              <li>
+<var>verified</var>.<var>document</var> to the information that was checked and
+returned as successfully protected by the verification process for the securing
+mechanism
```

Comment:
It's not clear to me what this document would be in either DI or vc-jose-cose. I would suggest adding an example of this. 

Specifically:
* When using the DI mechanism, I don't know if `verified.document` must have the `proof` element, or if it should be canonnicalized.
* When using the vc-jose-cose, I don't know if the `verified.document` should include the protected header.

I think examples would help here.

Reply:
Good question. In the DI case, it's supposed to be the conforming document, which is either the (verified) verifiable credential or the verifiable presentation (in its entirety). 

In the JOSE-COSE case, we have a problem because VC-JWT-1.1 encapsulated properties, and the new form doesn't... and it's not possible to re-verify the contents unless you have the entire payload, so `document` might have to be an `application/jwt`.

The result we want is that, whatever is handed back, is safe to use by the calling application.